sztp-agent/pkg/secureAgent/daemon.go

```diff
@@ -103,8 +113,47 @@ func (a *Agent) doRequestBootstrapServer() error {
 		return derr
 	}
 	res.IetfSztpBootstrapServerOutput.ConveyedInformation = string(data.Bytes)
+	a.BootstrapServerOnboardingInfo = oi
 	log.Println(res)
-	// TODO: download and verify OS image
-	log.Println(oi.IetfSztpConveyedInfoOnboardingInformation.BootImage.DownloadURI)
+	return nil
+}
+
+func (a *Agent) downloadAndValidateImage() error {
+	log.Printf("[INFO] Starting the Download Image: %v", a.BootstrapServerOnboardingInfo.IetfSztpConveyedInfoOnboardingInformation.BootImage.DownloadURI)
+	// Download the image from DownloadURI and save it to a file
+	a.BootstrapServerOnboardingInfo.IetfSztpConveyedInfoOnboardingInformation.InfoTimestampReference = fmt.Sprintf("%+8d", time.Now().Unix())
+	var wg sync.WaitGroup
+	wg.Add(len(a.BootstrapServerOnboardingInfo.IetfSztpConveyedInfoOnboardingInformation.BootImage.DownloadURI))
+	for i, item := range a.BootstrapServerOnboardingInfo.IetfSztpConveyedInfoOnboardingInformation.BootImage.DownloadURI {
+		//goroutine to make the download concurrent
```

Comment:
the download should not be concurrent here... 
I think the SPEC intent is if you fail to download first location, then move to second location.... 
if first image was downloaded and matched the SHA, you break and not continue to next URL...